controllers/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -128,6 +133,81 @@ func (r *NodeNetworkConfigurationPolicyReconciler) initializeEnactment(policy nm
 	})
 }
 
+func (r *NodeNetworkConfigurationPolicyReconciler) getEnactmentCount(policy *nmstatev1beta1.NodeNetworkConfigurationPolicy) (enactmentconditions.ConditionCount, error) {
+	enactments := nmstatev1beta1.NodeNetworkConfigurationEnactmentList{}
+	policyLabelFilter := client.MatchingLabels{nmstateapi.EnactmentPolicyLabel: policy.GetName()}
+	err := r.Client.List(context.TODO(), &enactments, policyLabelFilter)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting enactment list failed")
+	}
+	enactmentCount := enactmentconditions.Count(enactments, policy.Generation)
+	return enactmentCount, nil
+}
+
+func (r *NodeNetworkConfigurationPolicyReconciler) setNodeRunningUpdate(policyKey types.NamespacedName) error {
+	policy := &nmstatev1beta1.NodeNetworkConfigurationPolicy{}
+	err := r.Client.Get(context.TODO(), policyKey, policy)
+	if err != nil {
+		return err
+	}
+	if policy.Status.NodeRunningUpdate != "" {
+		return fmt.Errorf("another node is working on configuration")
+	}
+	policy.Status.NodeRunningUpdate = nodeName
+	policy.Status.NodeUpdateStart = &metav1.Time{Time: time.Now()}
+	err = r.Client.Status().Update(context.TODO(), policy)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *NodeNetworkConfigurationPolicyReconciler) claimNodeRunningUpdate(
+	policy *nmstatev1beta1.NodeNetworkConfigurationPolicy,
+	ec *enactmentconditions.EnactmentConditions,
+) (ctrl.Result, error) {
+	if policy.Status.NodeRunningUpdate != "" && metav1.Now().Sub(policy.Status.NodeUpdateStart.Time) > nodeConfigurationTimeout {
+		// a node has been running the update for too long
+		errmsg := fmt.Errorf("A node has been configuring for too long, aborting")
+		ec.NotifyFailedToConfigure(errmsg)
+		return reconcile.Result{}, errmsg
+	}
+	err := r.setNodeRunningUpdate(types.NamespacedName{Name: policy.GetName(), Namespace: policy.GetNamespace()})
+	if err != nil {
+		return ctrl.Result{RequeueAfter: nodeRunningUpdateRetryTime}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *NodeNetworkConfigurationPolicyReconciler) releaseNodeRunningUpdate(policyKey types.NamespacedName) {
+	instance := &nmstatev1beta1.NodeNetworkConfigurationPolicy{}
+	_ = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		err := r.Client.Get(context.TODO(), policyKey, instance)
+		if err != nil {
+			r.Log.Info("Failed to get policy, retrying")
+			return err
+		}
+		// release only if we are the owner
+		if instance.Status.NodeRunningUpdate != nodeName {
+			return nil
+		}
```

Comment:
Remove this if we move `releaseNodeRunningUpdate` to the proper place withing Reconcile

Reply:
Done